src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -331,6 +331,33 @@ partition_downloader::download_log(const remote_manifest_path& manifest_key) {
     co_return result;
 }
 
+void partition_downloader::update_downloaded_offsets(
```

Comment:
nit: looks like this function doesn't need the whole `dlpart` and should just return the resulting offset range. And the name then should be something like `determine_consecutive_offset_range`